patches/disable-downloads.diff

```diff
@@ -0,0 +1,150 @@
++import { SplitEditorsVertically, InEditorZenModeContext, ActiveEditorCanRevertContext, ActiveEditorGroupLockedContext, ActiveEditorCanSplitInGroupContext, SideBySideEditorActiveContext, AuxiliaryBarVisibleContext, SideBarVisibleContext, PanelAlignmentContext, PanelMaximizedContext, PanelVisibleContext, ActiveEditorContext, EditorsVisibleContext, TextCompareEditorVisibleContext, TextCompareEditorActiveContext, ActiveEditorGroupEmptyContext, MultipleEditorGroupsContext, EditorTabsVisibleContext, IsCenteredLayoutContext, ActiveEditorGroupIndexContext, ActiveEditorGroupLastContext, ActiveEditorReadonlyContext, EditorAreaVisibleContext, ActiveEditorAvailableEditorIdsContext, DirtyWorkingCopiesContext, EmptyWorkspaceSupportContext, EnterMultiRootWorkspaceSupportContext, HasWebFileSystemAccess, IsFullscreenContext, OpenFolderWorkspaceSupportContext, RemoteNameContext, VirtualWorkspaceContext, WorkbenchStateContext, WorkspaceFolderCountContext, PanelPositionContext, IsEnabledFileDownloads } from 'vs/workbench/common/contextkeys';
+ import { TEXT_DIFF_EDITOR_ID, EditorInputCapabilities, SIDE_BY_SIDE_EDITOR_ID, DEFAULT_EDITOR_ASSOCIATION } from 'vs/workbench/common/editor';
+ import { trackFocus, addDisposableListener, EventType } from 'vs/base/browser/dom';
+ import { preferredSideBySideGroupDirection, GroupDirection, IEditorGroupsService } from 'vs/workbench/services/editor/common/editorGroupsService';
+@@ -194,6 +194,9 @@ export class WorkbenchContextKeysHandler
+ 		this.auxiliaryBarVisibleContext = AuxiliaryBarVisibleContext.bindTo(this.contextKeyService);
+ 		this.auxiliaryBarVisibleContext.set(this.layoutService.isVisible(Parts.AUXILIARYBAR_PART));
+ 
++		// code-server
++		IsEnabledFileDownloads.bindTo(this.contextKeyService).set(this.environmentService.isEnabledFileDownloads)
++
+ 		this.registerListeners();
+ 	}
+ 
+Index: code-server/lib/vscode/src/vs/workbench/services/environment/common/environmentService.ts
+===================================================================
+--- code-server.orig/lib/vscode/src/vs/workbench/services/environment/common/environmentService.ts
++++ code-server/lib/vscode/src/vs/workbench/services/environment/common/environmentService.ts
+@@ -44,6 +44,9 @@ export interface IWorkbenchEnvironmentSe
+ 	readonly filesToOpenOrCreate?: IPath[] | undefined;
+ 	readonly filesToDiff?: IPath[] | undefined;
+ 
++	// --- code-server
++	readonly isEnabledFileDownloads: boolean;
++
```

Comment:
I added it to the common `environmentService` since this seems to be the `environmentService` used in `contextkeys` [here](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/browser/contextkeys.ts#L78) but that leads to type errors since I didn't add it to the other `environmentService`s. See here:
https://github.com/coder/code-server/runs/5796104643?check_suite_focus=true#step:11:17

So either:
1. I need to update the other `environmentService`s
2. I need to take a different approach
3. something else

Reply:
@code-asher thoughts on this?

Reply:
Definitely will comment, I just wanted to get the missing context key to see how it all ties together and what can be done

Reply:
All right so since the context key is only used in the browser I think keeping this value in only the browser environment service makes the most sense.

To do that:

1. Remove `isEnabledFileDownloads` from this file (`common/environmentService`)
2. In `browser/contextkeys` instead of using `IWorkbenchEnvironmentService` use `IBrowserWorkbenchEnvironmentService` (replace in the constructor and the import).
3. Add `isEnabledFileDownloads: boolean` to `IBrowserWorkbenchEnvironmentService`

At this point the native environment service extends the browser environment service for some reason...seems like a bug to me so we could change it to not do that.  Or if that feels sketchy we could just add a `?` to `isEnabledFileDownloads`.

If we do the second option we also will need `=== true` to the `set` call in `browser/contextkeys` since it does not like the undefined.

Reply:
Although...maybe to minimize patching we could just add a `?` to this line here (plus the `=== true`)

Reply:
That seems to be how they do `remoteAuthority` which is web-only so even if that is not the most technically correct it does follow an existing pattern.

Reply:
nvm `remoteAuthority` is used in native as well, ignore me :stuck_out_tongue: 

But maybe for now we just go the minimal patch route and if we were to upstream we could do it more "properly"

Reply:
> At this point the native environment service extends the browser environment service for some reason...seems like a bug to me so we could change it to not do that

Yeah, that's what I was super confused about. Okay, I'm going to go this route! Thanks for elaborating and providing some steps on how we should tackle this 🙌🏼